packages/core/src/services/documents/scan.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path'
+
+import { readMetadata, Document as RefDocument } from '@latitude-data/compiler'
+import { ConversationMetadata, scan } from '@latitude-data/promptl'
+
+import { Commit, DocumentVersion, Workspace } from '../../browser'
+import {
+  LatitudeError,
+  Result,
+  TypedResult,
+  UnprocessableEntityError,
+} from '../../lib'
+import { DocumentVersionsRepository } from '../../repositories'
+
+/**
+ * This is an internal method. It should always receives
+ * workspaceId from a trusted source. Like for example API gateway that validates
+ * requested documents belongs to the right workspace.
+ */
+export async function scanDocumentContent({
+  workspaceId,
+  document,
+  commit,
+}: {
+  workspaceId: Workspace['id']
+  document: DocumentVersion
+  commit: Commit
+}): Promise<TypedResult<ConversationMetadata, LatitudeError>> {
+  const documentScope = new DocumentVersionsRepository(workspaceId)
+  const docs = await documentScope
+    .getDocumentsAtCommit(commit)
+    .then((r) => r.unwrap())
+
+  const docInCommit = docs.find((d) => d.documentUuid === document.documentUuid)
+
+  if (!docInCommit) {
+    return Result.error(
+      new UnprocessableEntityError('Document not found in commit', {}),
+    )
+  }
+
+  const referenceFn = async (
```

Comment:
I will use the same function. Please export so i can use it. Meantime I'll do my own